### PR TITLE
 Fix 999.9... edge case in ticker.EngFormatter for negative numbers 

### PR DESCRIPTION
**Pull request by Lewn**
## PR Summary
When using ticker.EngFormatter, a -999.999 value will be displayed as -1000 instead of -1k. The edge case detection for the positive numbers is now extended to the negative numbers.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -1268,7 +1268,9 @@ def format_eng(self, num):
         # Taking care of the cases like 999.9..., which may be rounded to 1000
         # instead of 1 k.  Beware of the corner case of values that are beyond
         # the range of SI prefixes (i.e. > 'Y').
-        if float(format(mant, fmt)) >= 1000 and pow10 < max(self.ENG_PREFIXES):
+        _fmant = float(format(mant, fmt))
+        if ((_fmant >= 1000 or _fmant <= -1000)
```
Comment (jklymak) [inline lib/matplotlib/ticker.py]: `abs(_fmant) >= 1000` good enough?

File: lib/matplotlib/ticker.py [hunk 2]
```diff
@@ -1268,7 +1268,9 @@ def format_eng(self, num):
         # Taking care of the cases like 999.9..., which may be rounded to 1000
         # instead of 1 k.  Beware of the corner case of values that are beyond
         # the range of SI prefixes (i.e. > 'Y').
-        if float(format(mant, fmt)) >= 1000 and pow10 < max(self.ENG_PREFIXES):
+        _fmant = float(format(mant, fmt))
```
Comment (timhoffm) [inline lib/matplotlib/ticker.py]: Is there a particular reason to mark the variable as private? It's local anyway. Also I'd use a more telling name like `rounded_mant`.

Comment (Lewn) [reply lib/matplotlib/ticker.py]: I followed @jklymak and changed the explicit positive and negative check to a absolute value check. This eliminates the variable completely.
